packages/@react-spectrum/picker/test/Picker.test.js

```diff
@@ -56,114 +63,1398 @@ describe('Picker', function () {
-        </Picker>
-      </Provider>
-    );
+  describe('opening', function () {
+    it('can be opened on mouse down', function () {
+      let onOpenChange = jest.fn();
+      let {getAllByRole, getByRole} = render(
+        <Provider theme={theme}>
+          <Picker label="Test" onOpenChange={onOpenChange}>
+            <Item>One</Item>
+            <Item>Two</Item>
+            <Item>Three</Item>
+          </Picker>
+        </Provider>
+      );
 
-    let picker = getByRole('button');
-    expect(onOpenChangeSpy).toHaveBeenCalledTimes(0);
-    act(() => triggerPress(picker));
-    act(() => jest.runAllTimers());
-    expect(onOpenChangeSpy).toHaveBeenCalledTimes(1);
+      expect(getAllByRole('listbox')).toHaveLength(1);
 
-    let listbox = getAllByRole('listbox')[1]; // ignore the one in the background for now
-    let label = getAllByText('Test')[0];
-    expect(listbox).toBeVisible();
-    expect(listbox).toHaveAttribute('aria-labelledby', label.id);
-
-    let item1 = within(listbox).getByText('One');
-    let item2 = within(listbox).getByText('Two');
-    let item3 = within(listbox).getByText('Three');
-    expect(item1).toBeTruthy();
-    expect(item2).toBeTruthy();
-    expect(item3).toBeTruthy();
-
-    act(() => triggerPress(item3));
-    expect(onSelectionChange).toHaveBeenCalledTimes(1);
-    act(() => jest.runAllTimers());
-    expect(onOpenChangeSpy).toHaveBeenCalledTimes(2);
-    expect(getAllByRole('listbox').length).toBe(1);
-
-
-    act(() => triggerPress(picker));
-    act(() => jest.runAllTimers());
-    expect(onOpenChangeSpy).toHaveBeenCalledTimes(3);
-
-    listbox = getAllByRole('listbox')[1];
-    item1 = within(listbox).getByText('One');
-
-    // act callback must return a Promise or undefined, so we return undefined here
-    act(() => {
-      fireEvent.keyDown(item1, {key: 'Escape'});
-    });
-    expect(onSelectionChange).toHaveBeenCalledTimes(1); // still expecting it to have only been called once
-    act(() => jest.runAllTimers());
-    expect(onOpenChangeSpy).toHaveBeenCalledTimes(4);
-    expect(getAllByRole('listbox').length).toBe(1);
+      let picker = getByRole('button');
+      act(() => {fireEvent.mouseDown(picker);});
+      act(() => jest.runAllTimers());
+
+      let listbox = getAllByRole('listbox')[1]; // ignore the one in the background for now
+      expect(listbox).toBeVisible();
+      expect(onOpenChange).toBeCalledTimes(1);
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+      expect(picker).toHaveAttribute('aria-expanded', 'true');
+      expect(picker).toHaveAttribute('aria-controls', listbox.id);
+
+      let items = within(listbox).getAllByRole('option');
+      expect(items.length).toBe(3);
+      expect(items[0]).toHaveTextContent('One');
+      expect(items[1]).toHaveTextContent('Two');
+      expect(items[2]).toHaveTextContent('Three');
+
+      expect(document.activeElement).toBe(items[0]);
+    });
+
+    it('can be opened on touch up', function () {
```

Comment:
The touch verse mouse events seem to be difference between this one and the one above. Can these be combined? Especially since the next two are also basically the same, just different events to open.